src/Upgrader/Business/Upgrader/Upgrader.php

```diff
@@ -7,63 +7,75 @@
 
 class Upgrader implements UpgraderInterface
 {
     /**
-     * @var \Upgrader\Business\Upgrader\Bridge\ReleaseGroupTransferBridgeInterface
+     * @var \Upgrader\Business\Upgrader\Strategy\ComposerUpdateStrategy
      */
-    protected $releaseGroupManager;
+    protected $composerUpdateStrategy;
 
     /**
-     * @var \Upgrader\Business\Upgrader\Bridge\PackageManagementSystemBridge
+     * @var \Upgrader\Business\Upgrader\Strategy\ReleaseGroupStrategy
      */
-    protected $dataProviderManager;
+    protected $releaseGroupStrategy;
 
     /**
      * @var \Upgrader\Business\VersionControlSystem\VcsInterface
      */
     protected $vcs;
 
     /**
-     * @param \Upgrader\Business\Upgrader\Bridge\ReleaseGroupTransferBridgeInterface $releaseGroupManager
-     * @param \Upgrader\Business\Upgrader\Bridge\PackageManagementSystemBridge $dataProviderManager
+     * @param \Upgrader\Business\Upgrader\Strategy\ComposerUpdateStrategy $composerUpdateStrategy
+     * @param \Upgrader\Business\Upgrader\Strategy\ReleaseGroupStrategy $releaseGroupStrategy
      * @param \Upgrader\Business\VersionControlSystem\VcsInterface $vcs
      */
     public function __construct(
-        ReleaseGroupTransferBridgeInterface $releaseGroupManager,
-        PackageManagementSystemBridge $dataProviderManager,
+        ComposerUpdateStrategy $composerUpdateStrategy,
+        ReleaseGroupStrategy $releaseGroupStrategy,
         VcsInterface $vcs
     ) {
-        $this->releaseGroupManager = $releaseGroupManager;
-        $this->dataProviderManager = $dataProviderManager;
+        $this->composerUpdateStrategy = $composerUpdateStrategy;
+        $this->releaseGroupStrategy = $releaseGroupStrategy;
         $this->vcs = $vcs;
     }
 
     /**
+     * @param \Upgrader\Business\Upgrader\Request\UpgraderRequest $request
+     *
      * @return \Upgrader\Business\Upgrader\Response\Collection\UpgraderResponseCollection
      */
-    public function upgrade(): UpgraderResponseCollection
+    public function upgrade(UpgraderRequest $request): UpgraderResponseCollection
     {
         $responses = new UpgraderResponseCollection();
+
+        $checkResponse = $this->vcs->checkTargetBranchExists();
+        $responses->add($checkResponse);
+        if (!$checkResponse->isSuccess()) {
+            return $responses;
+        }
         $checkResponse = $this->vcs->checkUncommittedChanges();
         $responses->add($checkResponse);
         if (!$checkResponse->isSuccess()) {
             return $responses;
         }
-        $dataProviderResponse = $this->dataProviderManager->getNotInstalledReleaseGroupList();
-        $packageManagerResponses = $this->releaseGroupManager->requireCollection(
-            $dataProviderResponse->getReleaseGroupCollection()
-        );
-        $responses->addCollection($packageManagerResponses);
-        if (!$packageManagerResponses->hasSuccessfulResponse()) {
+
+        if ($request->getStrategyEnum()->isComposerUpdate()) {
```

Comment:
might be
$upgradeResponses = $this->strategy->upgrade();

Reply:
++